FinSearchUnified/Tests/Bundle/SearchBundleFindologic/NavigationQueryBuilderTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace FinSearchUnified\Tests\Bundle\SearchBundleFindologic;
+
+use Exception;
+use FinSearchUnified\Bundle\SearchBundleFindologic\NavigationQueryBuilder;
+use FinSearchUnified\Bundle\SearchBundleFindologic\QueryBuilder;
+use PHPUnit\Framework\Assert;
+use Shopware\Bundle\PluginInstallerBundle\Service\InstallerService;
+use Shopware\Components\HttpClient\GuzzleHttpClient;
+use Shopware\Components\HttpClient\Response;
+use Shopware\Components\Test\Plugin\TestCase;
+use Shopware_Components_Config;
+
+class NavigationQueryBuilderTest extends TestCase
+{
+    /**
+     * @var Shopware_Components_Config
+     */
+    private $config;
+
+    /**
+     * @var InstallerService
+     */
+    private $installerService;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->config = Shopware()->Container()->get('config');
+        $this->installerService = Shopware()->Container()->get('shopware_plugininstaller.plugin_manager');
+        Shopware()->Session()->offsetSet('isSearchPage', false);
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testEndpointUrl()
+    {
+        $plugin = $this->installerService->getPluginByName('FinSearchUnified');
+        $shopUrl = rtrim(Shopware()->Shop()->getHost(), '/');
+        $parameters = [
+            'outputAdapter' => 'XML_2.0',
+            'userip' => 'UNKNOWN',
+            'revision' => $plugin->getVersion(),
+            'shopkey' => $this->config->offsetGet('ShopKey')
+        ];
+        $aliveUrl = sprintf(
+            '%s/%s/%s?shopkey=%s',
+            QueryBuilder::BASE_URL,
+            $shopUrl,
+            QueryBuilder::ALIVE_ENDPOINT,
+            $parameters['shopkey']
+        );
+        $executeUrl = sprintf(
+            '%s/%s/%s?%s',
+            QueryBuilder::BASE_URL,
+            $shopUrl,
+            NavigationQueryBuilder::ENDPOINT,
+            http_build_query($parameters)
+        );
+        $httpResponse = new Response(200, [], 'alive');
+        $httpClientMock = $this->createMock(GuzzleHttpClient::class);
+        $aliveCallback = $this->returnCallback(function ($url) use ($aliveUrl, $httpResponse) {
+            Assert::assertSame($aliveUrl, $url);
```

Comment:
Since we are checking the provided parameter, this assertion should be made in the `with` part. So check if something like this is possible:
```
$httpClientMock->expects($this->exactly(2))
    ->method('get')
    ->with($this->onConsecutive([THE CALLBACKS GOE HERE]))
    ->will($this->onConsecutiveCalls($httpResponse, $httpResponse));
```
See  `\FinSearchUnified\Tests\Subscriber\FrontendTest::testBeforeSearchIndexAction` for reference.

If it turns out that this approach won't work, we can stick with the current implementation.

Reply:
Well, that's unfortunate but okay.